src/repl.jl

```diff
@@ -14,9 +14,17 @@ function get_main_mode()
   mode
 end
 
-isREPL() = isdefined(Base, :active_repl) &&
-           isdefined(Base.active_repl, :interface) &&
-           isdefined(Base.active_repl.interface, :modes)
+function isREPL(; before_run_repl = false)
+  isdefined(Base, :active_repl) &&
+  isdefined(Base.active_repl, :interface) &&
+  isdefined(Base.active_repl.interface, :modes) &&
+  if before_run_repl
```

Comment:
Why not just check that all the time? Both the `isdefined` and the `isa` call should be very cheap.

Reply:
This comment explains it ?
https://github.com/JunoLab/Atom.jl/pull/180#discussion_r325072523